library/core/tests/any.rs

```diff
@@ -166,40 +173,41 @@ impl Provider for SomeConcreteType {
 // Test the provide and request mechanisms with a by-reference trait object.
 #[test]
 fn test_provider() {
-    let obj: &dyn Provider = &SomeConcreteType { some_string: "hello".to_owned() };
+    let obj: &dyn std::error::Error = &SomeConcreteType { some_string: "hello".to_owned() };
 
-    assert_eq!(&**request_ref::<String>(obj).unwrap(), "hello");
-    assert_eq!(&*request_value::<String>(obj).unwrap(), "bye");
-    assert_eq!(request_value::<u8>(obj), None);
+    assert_eq!(&**obj.request_ref::<String>().unwrap(), "hello");
+    assert_eq!(&*obj.request_value::<String>().unwrap(), "bye");
+    assert_eq!(obj.request_value::<u8>(), None);
 }
 
 // Test the provide and request mechanisms with a boxed trait object.
 #[test]
 fn test_provider_boxed() {
-    let obj: Box<dyn Provider> = Box::new(SomeConcreteType { some_string: "hello".to_owned() });
+    let obj: Box<dyn std::error::Error> =
+        Box::new(SomeConcreteType { some_string: "hello".to_owned() });
 
-    assert_eq!(&**request_ref::<String>(&*obj).unwrap(), "hello");
-    assert_eq!(&*request_value::<String>(&*obj).unwrap(), "bye");
-    assert_eq!(request_value::<u8>(&*obj), None);
+    assert_eq!(&**obj.request_ref::<String>().unwrap(), "hello");
+    assert_eq!(&*obj.request_value::<String>().unwrap(), "bye");
+    assert_eq!(obj.request_value::<u8>(), None);
 }
 
 // Test the provide and request mechanisms with a concrete object.
 #[test]
 fn test_provider_concrete() {
     let obj = SomeConcreteType { some_string: "hello".to_owned() };
 
-    assert_eq!(&**request_ref::<String>(&obj).unwrap(), "hello");
-    assert_eq!(&*request_value::<String>(&obj).unwrap(), "bye");
-    assert_eq!(request_value::<u8>(&obj), None);
+    assert_eq!(&**(&obj as &dyn std::error::Error).request_ref::<String>().unwrap(), "hello");
+    assert_eq!(&*(&obj as &dyn std::error::Error).request_value::<String>().unwrap(), "bye");
+    assert_eq!((&obj as &dyn std::error::Error).request_value::<u8>(), None);
```

Comment:
Once you add the methods to the error trait as I previously suggested, this can be simplified.

Reply:
Even using the free functions that we ultimately landed on simplified this. Thanks for the suggestion!